test/unittest/blas2/blas2_gemv_test.cpp

```diff
@@ -145,6 +145,186 @@ TYPED_TEST(BLAS_Test, gemv_test_transposed) {
   ex.get_policy_handler().template deallocate<scalar_t>(v_c_gpu);
 }
 
+// Test for a non-unit beta value
+REGISTER_PREC(float, 1e-4, gemv_test_legacy_beta)
+REGISTER_PREC(double, 1e-8, gemv_test_legacy_beta)
+
+TYPED_TEST(BLAS_Test, gemv_test_legacy_beta) {
+  using scalar_t = typename TypeParam::scalar_t;
+  using ExecutorType = typename TypeParam::executor_t;
+  using TestClass = BLAS_Test<TypeParam>;
+  using test = class gemv_test_legacy_beta;
+
+  int m = 125;
+  int n = 127;
+  int lda = m;
+  int incX = 1;
+  int incY = 1;
+  const char* t_str = "n";  // Testing the no transpose matrix
+  scalar_t prec = TestClass::template test_prec<test>();
+  scalar_t alpha = scalar_t(1.0);
```

Comment:
in the main test you can set std::arry<scalar_t, 4>(0.0, 1.0, 4.0, 5.0); for alpha and beta, similarly for m, and n, you can look into the gemm_def.hpp if you want to test multiple sizes and values. In this way we can remove alot of code duplication

Reply:
My concern with this is that it hides the reasons the tests are failing. For example, if you can see that the "beta" tests are failing, it means that something to do with "beta" is wrong. I can see where code duplication could be an issue though.